src/types/UiaTextRangeBase.cpp

```diff
@@ -346,6 +347,59 @@ IFACEMETHODIMP UiaTextRangeBase::FindAttribute(_In_ TEXTATTRIBUTEID /*textAttrib
     return E_NOTIMPL;
 }
 
+IFACEMETHODIMP UiaTextRangeBase::FindText(_In_ BSTR text,
+                                          _In_ BOOL searchBackward,
+                                          _In_ BOOL ignoreCase,
+                                          _Outptr_result_maybenull_ ITextRangeProvider** ppRetVal) noexcept
+{
+    // TODO GitHub #1914: Re-attach Tracing to UIA Tree
+    //Tracing::s_TraceUia(this, ApiCall::FindText, nullptr);
+    RETURN_HR_IF(E_INVALIDARG, ppRetVal == nullptr);
+    *ppRetVal = nullptr;
+    try
+    {
+        const std::wstring wstr{ text, SysStringLen(text) };
+        const auto sensitivity = ignoreCase ? Search::Sensitivity::CaseInsensitive : Search::Sensitivity::CaseSensitive;
+
+        auto searchDirection = Search::Direction::Forward;
+        auto searchAnchor = _start;
+        if (searchBackward)
+        {
+            searchDirection = Search::Direction::Backward;
+            searchAnchor = _end;
+        }
+
+        Search searcher{ *_pData, wstr, searchDirection, sensitivity, searchAnchor };
+
+        HRESULT hr = S_OK;
+        if (searcher.FindNext())
+        {
+            const auto bufferSize = _pData->GetTextBuffer().GetSize();
+            const auto foundLocation = searcher.GetFoundLocation();
+            const auto start = foundLocation.first;
+
+            // we need to increment the position of end because it's exclusive
+            auto end = foundLocation.second;
+            bufferSize.IncrementInBounds(end, true);
+
+            // make sure what was found is within the bounds of the current range
+            if ((searchDirection == Search::Direction::Forward && bufferSize.CompareInBounds(end, _end) < 0) ||
+                (searchDirection == Search::Direction::Backward && bufferSize.CompareInBounds(start, _start) > 0))
+            {
+                hr = Clone(ppRetVal);
+                if (SUCCEEDED(hr))

```

Comment:
should this be `RETURN_IF_FAILED(Clone(ppRetVal))`?

Reply:
(is a failure to create a new UTR fatal for this function)

Reply:
what's the point in cloning, anyway? all we are is a start and end, right? i guess it helps us carry over other info..

Reply:
Clone copies over the parent, start, end, pData, and wordDelimiters. Clone also calls the UiaTextRange constructor (aside from the base one). So, if somebody tries to add extra data, this would be helpful.

Did a minor refactor to get rid of `hr` tho.